### DISCUSSION TITLE
Redirect URL setting after authentication is ignored

### DISCUSSION THREAD

**Discussion by yusakapon** [Web3Auth/Web3Auth#383] [Help]
I am implementing this using custom authentication.
The connection with web3auth is completely successful, but any redirects I specify after authentication are ignored.
(To be precise, I am redirected to the page where web3auth is initialized.)
Is there a wrong way to specify this? Or is it impossible to specify a redirect destination for custom authentication?

I have checked this document
https://web3auth.io/docs/sdk/web/openlogin#connectto

```js
const connecter = await web3Auth.connectTo(
    WALLET_ADAPTERS.OPENLOGIN,
    {
        loginProvider: "jwt",
        extraLoginOptions: {
            id_token: jwtToken, 
            domain: "https://xxxxx.amazoncognito.com/",
            verifierIdField: "email",
            redirect_uri: window.location.href,
        },
    }
);
```

**Comment by himanshuchawla009** [comment]
Hey, redirect_uri will work only in implicit oauth flow (when you won't be passing id token your self, this redirect_url means to specify redirect from oauth provider to your app), it will have no effects when you are passing jwt token your self. You can specify redirect_url (for redirect from openlogin to your app) in OpenloginAdapter constructor loginSettings params to redirect to your app's route after openlogin's 2fa.

**Reply by yusakapon** [reply]
@himanshuchawla009   
Thank you for your response.
Our website is configured to allow login on multiple pages.
The constructor of `OpenloginAdapter` executes only once, but `web3Auth.connectTo` is called on multiple pages.

For example, I can login on pages like `https://myapp.com/mypage` and `https://myapp.com/detail/111`.

If the constructor of `OpenloginAdapter` is executed on `~/mypage` and I login on `~/detail/111`, I want to redirect to `~/detail/111` after login.

In other words, I want to dynamically change the redirect destination.
Is there any way to achieve this?

**Reply by himanshuchawla009** [reply]
For your use case, instead of passing redirectUrl in constructor you can pass in connectTo function as shown below:-

```

async connectTo(WALLET_ADAPTERS.OPENLOGIN, {  redirectUrl:   "https://myapp.com/mypage"  })

```

I hope this answers your query. 

@yusakapon

**Reply by yusakapon** [reply]
@himanshuchawla009 

I have actually tried this method, but the settings seem to be ignored.  
Is this method also invalid when using jwtToken?

```js
const connecter = await web3Auth.connectTo(
    WALLET_ADAPTERS.OPENLOGIN,
    {
        loginProvider: "jwt",
        extraLoginOptions: {
            id_token: jwtToken, 
            domain: "https://xxxx.amazoncognito.com/", 
            verifierIdField: "email",
        },
        redirectUrl: window.location.href,
    }
);
```

**Reply by yashovardhan** [reply]
Hey @yusakapon 

Looks like you're passing the redirectUri in the wrong location. It should be within extraLoginOptions. Have a look at our SDK Reference: https://web3auth.io/docs/sdk/web/core/custom-authentication#extraloginoptions
